lib/optimize/SideEffectsFlagPlugin.js

```diff
@@ -243,7 +243,12 @@ class SideEffectsFlagPlugin {
 						);
 
 						logger.time("update dependencies");
-						for (const module of modules) {
+
+						const optimizedModules = new Set();
```

Comment:
Ah gotcha. It seems the rough idea here is that we want to topological sort this "incoming connection optimization" for output reproducibility?

Reply:
Yep! For output reproducibility, they can be in any order as long as it's consistent between runs. But for reproducible output that properly shakes out all modules marked side-effect-free, we do need modules to have their incoming connections sorted in topological order